src/assets/js/react/components/FontManager/FontManagerBody.js

```diff
@@ -0,0 +1,344 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import SearchBox from './SearchBox'
+import FontList from './FontList'
+import AddFont from './AddFont'
+import {
+  getCustomFontList,
+  addFont,
+  editFont,
+  validationError,
+  clearAddFontMsg,
+  clearDropzoneError
+} from '../../actions/fontManager'
+import Alert from '../Alert/Alert'
+import { toggleUpdateFont } from '../../utilities/toggleUpdateFont'
+
+export class FontManagerBody extends Component {
+  static propTypes = {
+    getCustomFontList: PropTypes.func.isRequired,
+    id: PropTypes.string,
+    loading: PropTypes.bool.isRequired,
+    fontList: PropTypes.arrayOf(PropTypes.object).isRequired,
+    msg: PropTypes.object.isRequired,
+    clearDropzoneError: PropTypes.func.isRequired,
+    clearAddFontMsg: PropTypes.func.isRequired,
+    editFont: PropTypes.func.isRequired,
+    validationError: PropTypes.func.isRequired,
+    addFont: PropTypes.func.isRequired,
+    history: PropTypes.object.isRequired
+  }
+
+  state = {
+    label: '',
+    fontId: '',
+    fontStyles: {
+      regular: '',
+      italics: '',
+      bold: '',
+      bolditalics: ''
+    },
+    kashida: 0,
+    validateLabel: true,
+    validateRegular: true
+  }
+
+  componentDidMount () {
+    this.props.getCustomFontList()
+  }
+
+  componentDidUpdate (prevProps, prevState, snapshot) {
+    const { id, fontList, msg, history } = this.props
+
+    if (prevProps.id !== id && id) {
+      this.handleRequestFontDetails()
+    }
+
+    if (prevProps.id !== id && !id) {
+      this.handleRemoveFontDetails()
+    }
+
+    if (prevProps.fontList !== fontList && id && fontList) {
+      this.handleRequestFontDetails()
+    }
+
+    if (prevProps.msg !== msg && msg.success && !id) {
```

Comment:
Going to need some comments explaining each of these conditions a bit more so it's clearer about what's happening